src/index.ts

```diff
@@ -75,76 +76,62 @@ export function ruleFactory<
     const getReturnValue = () => results.lastValue;
 
     const startTime = performance.now();
-    if (trace) logTrace({ operation: 'begin', startTime });
+    logTrace({ operation: 'begin', startTime });
```

Comment:
The `if (trace)` is handled by `logTrace`.

Reply:
Usually the reason I do this is to avoid any expensive processing (`stringify`) that is passed to `logTrace`.
That doesn't appear to be the case here. 👍